### PR TITLE
Remove usage of RSVP.

### PR DESCRIPTION
**Pull request by rwjblue**
* Rewrite all usages of RSVP to use native promise
* Use [promise.hash.helper](https://github.com/stefanpenner/promise.hash.helper)   as a `RSVP.hash` replacement
* Remove `promise.prototype.finally` (in favor of native promise `.finally`)

Closes https://github.com/ember-cli/ember-cli/pull/8567

### CODE REVIEW

File: lib/utilities/hash.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = async function hash(obj) {
```
Comment (stefanpenner) [inline lib/utilities/hash.js]: Tests? Should we extract? If so, I can gladly do so today.

Comment (rwjblue) [reply lib/utilities/hash.js]: @stefanpenner - Ya, I can write some tests for the util. I was wondering if we should think about a larger refactor in RSVP itself to make RSVP use the native promise if possible...

**Comment by stefanpenner** [comment]
@rwjblue https://github.com/stefanpenner/promise.hash.helper/

File: lib/utilities/hash.js (hunk 1 above)
Comment (rwjblue) [reply lib/utilities/hash.js]: @stefanpenner extracted into [promise.hash.helper](https://github.com/stefanpenner/promise.hash.helper)

**Pull request merged**